datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -903,6 +904,36 @@ pub fn to_substrait_rex(
                 bounds,
             ))
         }
+        Expr::Like(Like {
+            negated,
+            expr,
+            pattern,
+            escape_char,
+        }) => make_substrait_like_expr(
+            false,
+            *negated,
+            expr,
+            pattern,
+            *escape_char,
+            schema,
+            col_ref_offset,
+            extension_info,
+        ),
+        Expr::ILike(Like {
+            negated,
+            expr,
+            pattern,
+            escape_char,
+        }) => make_substrait_like_expr(
+            true,
+            *negated,
+            expr,
+            pattern,
+            *escape_char,
+            schema,
+            col_ref_offset,
```

Comment:
[For future improvement. **Not completely related to this PR**]

I think having to carry around `col_ref_offset` for any expression-related functions unnecessarily overcrowds the code. Once we have `SubqueryAlias` support implemented, this should not be necessary anymore. I'll refactor the code when that happens.